mmdet3d/models/roi_heads/bbox_heads/centerpoint_bbox_head.py

```diff
@@ -0,0 +1,293 @@
+                - cls (torch.Tensor): [N, num_class]
+                - reg (torch.Tensor): [N, code_size]
+        """
+        pred_res = []
+
+        for i in range(len(roi_features)):
+            pred_res_batch = dict()
+            shared_features = self.shared_fc_layer(roi_features[i])
+            cls = self.cls_layers(shared_features)
+            reg = self.reg_layers(shared_features)
+            pred_res_batch.update(cls=cls)
+            pred_res_batch.update(reg=reg)
+            pred_res.append(pred_res_batch)
+
+        return pred_res
+
+    def loss(self, roi_features_sampled, sample_results, cfg):
+        """Loss function for CenterPoint bbox head.
+
+        Args:
+            roi_features_sampled list[torch.Tensor]: roi features after
+                sampling with shape of [N, 5*C]. cat([pos_box, neg_box])
+            sample_results list[(:obj:`SamplingResult`)]: Sampled results used
+                for training.
+            cfg (:obj:`ConfigDict`): Training config.
+
+        Returns:
+            dict: Losses from CenterPointBBoxHead.
+        """
+        losses = dict()
+
+        pred_res = self(roi_features_sampled)
+
+        label, bbox_targets, pos_gt_bboxes, reg_mask, label_weights, \
+            bbox_weights = self.get_targets(sample_results, cfg)
+
+        cls_pred = torch.cat([pred_batch['cls'] for pred_batch in pred_res],
+                             dim=0).view(-1)
+        label = torch.cat(list(label), dim=0).view(-1)
+        label_weights = torch.cat(list(label_weights), dim=0)
+        loss_cls = self.loss_cls(cls_pred, label, label_weights)
+        losses.update(loss_cls=loss_cls)
+
+        bbox_pred = torch.cat([pred_batch['reg'] for pred_batch in pred_res],
+                              dim=0)
+        bbox_targets = torch.cat(
+            list(bbox_targets), dim=0).reshape(-1, self.code_size)
+        bbox_weights = torch.cat(list(bbox_weights), dim=0).reshape(-1, 1)
+        loss_reg = self.loss_reg(bbox_pred, bbox_targets, bbox_weights)
+        losses.update(loss_reg=loss_reg)
+
+        return losses
+
+    def get_bboxes(self, roi_features, img_metas, rois):
+        """Generate bboxes from one-stage rois and bbox head predictions.
+
+        Args:
+            roi_features list[torch.Tensor]: Extracted features. The shape
+                of each roi feature is [N, 5*C]
+            imput_metas (list[dict]): Meta info of each input.
+            rois (list[list[bboxes, scores, labels]]): Decoded bbox, scores
+                 and labels.
+                - bboxes (:obj:`BaseInstance3DBoxes`): Prediction bboxes
+                    after nms
+                - scores (torch.Tensor): Prediction scores after nms with the
+                    shape of [N].
+                - labels (torch.Tensor): Prediction labels after nms with the
+                    shape of [N].
+
+        Returns:
+            list[list[boxes, scores, labels]]: final predicted bboxes
```

Comment:
list[list[tensor]]: Final predicted results, the outer list indicates different images, and the internal list contains three tensors, arrange as the following three tensors

                    - bboxes (:obj:`BaseInstance3DBoxes`): finale bboxes
                    - scores (torch.Tensor): finale scores
                    - labels (torch.Tensor): finale labels